### PR TITLE
Add more convenience functions to Player class

### PR DESCRIPTION
**Pull request by tobbi**
Waddup, biatches! I just added a few convenience methods to the player class that will reduce some complexity in if statements elsewhere.

Have fun "reviewing" (aka tearing into my code changes).

### CODE REVIEW

**Comment by MatusGuy** [comment]
no

**Comment by tobbi** [comment]
Thank you, @MatusGuy for your enthusiasm in reviewing the PR.

**Comment by mrkubax10** [comment]
I agree with Tobbi, could you try being a little bit nicer?

**Comment by tobbi** [comment]
It's an ongoing feud at this point. TBH, I don't know how serious either of us are about this at this point. I just know that I'm more on edge / unhinged than usual.

File: src/object/player.hpp [hunk 1]
```diff
@@ -135,6 +135,18 @@ class Player final : public MovingObject
...
+   * Returns true if the player is currently alive 
+   * (not dying or dead)
+   */
+  bool is_alive() const { return !is_dying() && !is_dead(); }
+  
+  /**
+   * Returns true if the player can be controlled
+   * by the player (alive and not currently in a win sequence)
+   */
+  bool is_active() const { return is_alive() && !is_winning(); }
```
Comment (MatusGuy) [inline src/object/player.hpp:148]:
```suggestion
  /**
   * Returns true if the player can be controlled.
   * (alive and not currently in a win sequence)
   */
  bool is_active() const { return is_alive() && !is_winning(); }
```

I just think repeating the word "player" while meaning two different things doesn't sound well and could probably cause confusion.

**Review by MatusGuy — APPROVED** [review]

**Comment by tobbi** [comment]
> > I agree with Tobbi, could you try being a little bit nicer?
> 
> I don't think his attitude was correct either. By saying "no", I was joking around, like he was in his PR description.
> 
> Anyway, I hope we can put that aside.

It was quite clear from the start that we were both joking, that's why I edited your message to a very polite one, before you added it back. No worries. This is all fine, unlike some other things I could tell you about.

**Review by bruhmoent — APPROVED** [review]
Not tested.

**Pull request merged**